pubsub/gcp/gcp.go

```diff
@@ -163,7 +173,19 @@ func (p *publisher) PublishRaw(ctx context.Context, key string, m []byte) error
 		Data:       m,
 		Attributes: map[string]string{"key": key},
 	})
-	_, err := res.Get(ctx)
+	var err error
+	if p.topic.PublishSettings.DelayThreshold == 0 {
+		_, err = res.Get(ctx)
+	} else {
+		// if the DelayThreshold > 0, we use a goroutine
```

Comment:
There are two issues with this approach
1) you don't handle/surface the error 
2) you possibly lose messages messages if your process shuts down while Get() is blocking for the delay

I don't have a good suggestion right now how to improve this but I would think you need a way to track outstanding publishes.

Reply:
Is this perhaps just a matter of documentation? The application using this option would want to make sure it shuts down gracefully allowing for an interval equivalent to the value of DelayThreshold, but that's not something we can control here. I'm adding a note to that effect in the config comments.

Reply:
Documenting this issue is definitely a start but it still doesn't address the problem of not surfacing the errors. Using `log.Printf()` as you do right now isn't playing nice with the existing logging setup.

Reply:
I see what you're saying. I'm updating it to use the logger created for the pubsub package.